sdv/sequential/par.py

```diff
@@ -88,6 +113,107 @@ def get_parameters(self):
 
         return instantiated_parameters
 
-    def get_metadata(self):
-        """Return the ``SingleTableMetadata`` for this synthesizer."""
-        return self.metadata
+    def preprocess(self, data):
+        """Transform the raw data to numerical space.
+
+        For PAR, none of the sequence keys are transformed.
+
+        Args:
+            data (pandas.DataFrame):
+                The raw data to be transformed.
+
+        Returns:
+            pandas.DataFrame:
+                The preprocessed data.
+        """
+        sequence_key_transformers = {sequence_key: None for sequence_key in self._sequence_key}
+        if self._data_processor._hyper_transformer.field_transformers == {}:
+            self.auto_assign_transformers(data)
+
+        self.update_transformers(sequence_key_transformers)
+        return super().preprocess(data)
+
+    def _fit_context_model(self, transformed):
+        LOGGER.debug(f'Fitting context synthesizer {self._context_synthesizer.__class__.__name__}')
+        if self.context_columns:
+            context = transformed[self._sequence_key + self.context_columns]
+        else:
+            context = transformed[self._sequence_key].copy()
+            # Add constant column to allow modeling
+            context[str(uuid.uuid4())] = 0
+
+        context = context.groupby(self._sequence_key).first().reset_index()
+        self._context_synthesizer.fit(context)
+
+    def _transform_sequence_index(self, sequences):
+        sequence_index_idx = self._data_columns.index(self._sequence_index)
+        for sequence in sequences:
+            data = sequence['data']
+            sequence_index = data[sequence_index_idx]
+            diffs = np.diff(sequence_index).tolist()
+            data[sequence_index_idx] = diffs[0:1] + diffs
+            data.append(sequence_index[0:1] * len(sequence_index))
+
+    def _fit_sequence_columns(self, timeseries_data):
+        self._model = PARModel(**self._model_kwargs)
+
+        # handle output name from rdt
+        if self._sequence_index:
+            modified_name = self._sequence_index + '.value'
```

Comment:
we don't add .value anymore?

Reply:
I'll get rid of this when we upgrade the rdt version to the one with the changes